Drop root privledges

bush6 opened the pull request:
I have added a feature allowing users to change uid and gid after starting as root. this is particulary handy for binding ports below 1024.

Commit:
added a feature to change uid/gid after startup to bind ports lower than
1024 on Linux

Commit:
Forgot the new file :o

calzoneman commented on lib/setuid.js:
```diff
@@ -0,0 +1,15 @@
+var Config = require("./config");
+
+if (Config.get("setuid.enabled")) {
+    setTimeout(function() {
```
Why is this in a setTimeout?

bush6 replied:
because if it isn't it drops the uid before the ports are bound

calzoneman replied:
How did you determine the timeout?  My concern would be that if the port bind happened to take uncharacteristically long, it could be a tricky nondeterministic issue to debug.

The ideal solution would be to add the callback to `http.createServer` etc. so that the setuid happens after the port finishes binding.  I'm not sure if there's a clean way to do this for multiple servers, besides keeping track of a counter variable or something.

bush6 replied:
I determined the timeout by tring it on my old 1.6ghz single thread laptop. I think 15ms should be long enough to bind ports. if 15ms is long enough on a 10 year old laptop then I think it should be right.

Commit:
set uid timeout

Allow the timeout to be changed allowing more time before dropping root
privledges

Commit:
Merge remote-tracking branch 'upstream/3.0' into 3.0

ghost commented:
I have made it so it is possible to change the timeout before it drops root.
